### PR TITLE
[HEAP-9795] Reset Identity

### PR DESCRIPTION
**Pull request by jmtaber129**
Add the `resetIdentity` method to the bridge + add tests.

### CODE REVIEW

**Comment by daviscabral** [comment]
Hey guys, not sure if this is to fix the ScrollView error, but now it does not trigger the error anymore but crashes the Heap lib with an exception causing just an warning (what I believe it's much better for now). 

Screenshot 2019-08-06 10 32 25

File: e2e/basics.spec.js [hunk 1]
```diff
@@ -146,6 +153,23 @@ describe('Basic React Native and Interaction Support', () => {
       assert(res.length).equal(1);
       assert(res[0]['i']).equal('foobar');
     });
+
+    it('should reset identity', async () => {
+      const { err1, err2, res1, res2 } = await new Promise(resolve => {
+        testUtil.findEventInRedisRequests({ t:  'pressInTestEvent2'}, (err1, res1) => {
```
Comment (GrandPoohBear) [inline e2e/basics.spec.js]: It's a little hard to follow what's happening here... Certainly, there's a lower threshold for readability, since this is test code, but perhaps factoring out a `findEventAfterSentinel` method or something would make it clearer?

File: e2e/basics.spec.js [hunk 2]
```diff
@@ -146,6 +153,23 @@ describe('Basic React Native and Interaction Support', () => {
...
+            resolve({ err1, err2, res1, res2 });
+          });
+        });
+      });
+
+      assert.not.exist(err1);
+      assert.not.exist(err2);
+      assert(res1.length).not.equal(0);
+      assert(res2.length).not.equal(0);
+      assert.not.exist(res2[0]['i']);
```
Comment (GrandPoohBear) [inline e2e/basics.spec.js]: Would you mind adding a comment as to what these keys are, and what you're actually checking here?

File: e2e/basics.spec.js [hunk 3]
```diff
@@ -269,7 +293,7 @@ describe('Basic React Native and Interaction Support', () => {
   describe('Autotrack', () => {
     it("should autotrack 'TouchableOpacity's", async () => {
       const expectedHierarchy =
-        'AppContainer;|App;|Provider;|HeapNavigationWrapper;|NavigationContainer;|Navigator;|NavigationView;|TabNavigationView;|ScreenContainer;|ResourceSavingScene;[key=Basics];|SceneView;|Connect(BasicsPage);|BasicsPage;|TouchableOpacity;[testID=touchableOpacityText];|';
+        'AppContainer;|App;|Provider;|HeapNavigationWrapper;|NavigationContainer;|Navigator;|NavigationView;|TabNavigationView;|ScreenContainer;|ResourceSavingScene;[key=Basics];|SceneView;|Connect(BasicsPage);|BasicsPage;|ScrollView;|TouchableOpacity;[testID=touchableOpacityText];|';
```
Comment (GrandPoohBear) [inline e2e/basics.spec.js]: Maybe it's time to factor a large prefix of this string into a constant and share it between the tests?  Sucks to have to modify _n_ tests for every minor UI change in TestDriver...

**Comment by GrandPoohBear** [comment]
@daviscabral - this particular change is _not_ to address that issue, but it's on our radar, and we're actively working on getting a fix for that.  We'll be sure to flag it for you when it's fixed!

**Comment by daviscabral** [comment]
@GrandPoohBear no worries, thanks, waiting on that to proceed with complete integration on product here.

File: e2e/basics.spec.js (hunk 1 above)
Comment (jmtaber129) [reply e2e/basics.spec.js]: Could you elaborate on what you mean by `findEventAfterSentinel` and what you'd like pulled out here?

File: e2e/basics.spec.js (hunk 3 above)
Comment (jmtaber129) [reply e2e/basics.spec.js]: Yea, I agree, although this should probably happen in a follow-up PR

File: e2e/basics.spec.js (hunk 1 above)
Comment (GrandPoohBear) [reply e2e/basics.spec.js]: I just mean that you have to jump through a couple of hoops to get what you want here, which makes this section particular hard to grok for the uninitiated.  It seemed to me like what you were really saying was "find this event after we're sure we've seen the sentinel".  Pulling that out into a named function might make this a lot easier to read for someone who's new to the project.

Comment (jmtaber129) [reply e2e/basics.spec.js]: All of these assertions happen after the sentinel anyway (since the `before` hook blocks until we see the sentinel).  I assume you mean before+after the `resetIdentity` call?

I don't think it's worth pulling this into a separate function.  Added comments for clarifying what this is doing
